lib/grape-swagger.rb

```diff
@@ -191,8 +191,9 @@ def parse_params(params, path, method)
               non_nested_parent_params.map do |param, value|
                 items = {}
 
-                raw_data_type = value.is_a?(Hash) ? (value[:type] || 'string').to_s : 'string'
-                data_type     = case raw_data_type
+                raw_data_type = value[:type] if value.is_a?(Hash)
```

Comment:
Nitpicking, since you're doing a `to_s` below you could:

```ruby
raw_data_type = value[:type] if value.is_a?(Hash)
raw_data_type = raw_data_type ? raw_data_type.to_s : 'string'
```